localstack/testing/snapshots/prototype.py

```diff
@@ -138,9 +138,11 @@ def match(self, key: str, obj: dict) -> None:
 
         self.called_keys.add(key)
 
-        self.observed_state[
-            key
-        ] = obj  # TODO: track them separately since the transformation is now done *just* before asserting
+        # order the obj to guarantee reference replacement works as expected
+        self.observed_state[key] = (
+            self._order_dict(obj) if isinstance(obj, dict) else obj
```

Comment:
nit: I think you don't need the first `isinstance` anymore as it checks it in the function now